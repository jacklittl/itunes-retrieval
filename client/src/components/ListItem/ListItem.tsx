import React from "react";
import { Result } from 'declarations/types';

import { 
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Link,
  Typography
} from '@material-ui/core';

type Props = {
  result: Result;
};

const ListItem = ({ result }: Props): JSX.Element => (

  <Card>
  
    <CardActionArea>

      <Link href={result.trackViewUrl} target="_blank" rel="noopener noreferrer">

        <CardMedia
          component="img"
          image={result.artworkUrl100}
          title={result.trackName}
          alt={result.trackName}
        />

        <CardContent>

          <Typography gutterBottom variant="h6" color="primary" component="h2">
            {result.trackName}
          </Typography>

          <Typography gutterBottom variant="body2" color="textSecondary" component="p">
            {result.artistName}
          </Typography>

          <Typography variant="body2" color="textSecondary" component="p">
            {result.collectionName}
          </Typography>

        </CardContent>

      </Link>

    </CardActionArea>

  </Card>

)

export default ( ListItem )