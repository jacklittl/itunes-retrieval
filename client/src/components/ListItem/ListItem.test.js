import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

import ListItem from './ListItem';

configure({adapter: new Adapter()});

const result = {
  artistId: 6906197,
  artistName: "Foo Fighters",
  artistViewUrl: "https://music.apple.com/us/artist/foo-fighters/6906197?uo=4",
  artworkUrl30: "https://is2-ssl.mzstatic.com/image/thumb/Music/v4/e4/23/35/e4233532-42dc-7ece-d18b-b9b2733cb4e6/source/30x30bb.jpg",
  artworkUrl60: "https://is2-ssl.mzstatic.com/image/thumb/Music/v4/e4/23/35/e4233532-42dc-7ece-d18b-b9b2733cb4e6/source/60x60bb.jpg",
  artworkUrl100: "https://is2-ssl.mzstatic.com/image/thumb/Music/v4/e4/23/35/e4233532-42dc-7ece-d18b-b9b2733cb4e6/source/100x100bb.jpg",
  collectionCensoredName: "In Your Honor",
  collectionExplicitness: "notExplicit",
  collectionId: 300855107,
  collectionName: "In Your Honor",
  collectionPrice: 16.99,
  collectionViewUrl: "https://music.apple.com/us/album/best-of-you/300855107?i=300855110&uo=4",
  country: "USA",
  currency: "USD",
  discCount: 2,
  discNumber: 1,
  isStreamable: true,
  kind: "song",
  previewUrl: "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview71/v4/a1/34/a5/a134a568-a74a-042e-2c7e-1192efbac8aa/mzaf_8504368996424262545.plus.aac.p.m4a",
  primaryGenreName: "Alternative",
  releaseDate: "2005-05-30T07:00:00Z",
  trackCensoredName: "Best of You",
  trackCount: 10,
  trackExplicitness: "notExplicit",
  trackId: 300855110,
  trackName: "Best of You",
  trackNumber: 3,
  trackPrice: 1.29,
  trackTimeMillis: 255261,
  trackViewUrl: "https://music.apple.com/us/album/best-of-you/300855107?i=300855110&uo=4",
  wrapperType: "track"
}

describe('List Item Render', () => {
  test('List Item Renders', () => {
    const component = renderer.create(
        <ListItem
            key={result.trackId}
            result={result}
        />
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});