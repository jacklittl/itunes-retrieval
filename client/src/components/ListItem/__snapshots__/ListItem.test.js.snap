// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Render List Item Renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="https://music.apple.com/us/album/best-of-you/300855107?i=300855110&uo=4"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Best of You"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        src="https://is2-ssl.mzstatic.com/image/thumb/Music/v4/e4/23/35/e4233532-42dc-7ece-d18b-b9b2733cb4e6/source/100x100bb.jpg"
        title="Best of You"
      />
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
        >
          Best of You
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Foo Fighters
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          In Your Honor
        </p>
      </div>
    </a>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;

exports[`List Item Render snapshot renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="https://music.apple.com/us/album/best-of-you/300855107?i=300855110&uo=4"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Best of You"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        src="https://is2-ssl.mzstatic.com/image/thumb/Music/v4/e4/23/35/e4233532-42dc-7ece-d18b-b9b2733cb4e6/source/100x100bb.jpg"
        title="Best of You"
      />
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
        >
          Best of You
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Foo Fighters
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          In Your Honor
        </p>
      </div>
    </a>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
